nexus/test-utils/src/background.rs

```diff
@@ -109,40 +118,40 @@ pub async fn activate_background_task(
 
             // Wait until the task has actually run and then is idle
             if matches!(&task.current, CurrentStatus::Idle) {
-                let current_activate = most_recent_activate_time(&task);
-                match (current_activate, last_activate) {
-                    (None, None) => {
-                        // task is idle but it hasn't started yet, and it was
-                        // never previously activated
+                match (&previous_task.last, &task.last) {
+                    (
+                        LastResult::NeverCompleted,
+                        LastResult::NeverCompleted,
+                    ) => {
+                        // task hasn't started yet
                         Err(CondCheckError::<()>::NotYet)
                     }
 
-                    (Some(_), None) => {
-                        // task was activated for the first time by this
-                        // function call, and it's done now (because the task is
-                        // idle)
+                    // task was activated for the first time by this function
+                    // call (or a concurrent one!), and it's done now (because
+                    // the task is idle)
+                    (LastResult::NeverCompleted, LastResult::Completed(_)) => {
                         Ok(task)
                     }
 
-                    (None, Some(_)) => {
-                        // the task is idle (due to the check above) but
-                        // `most_recent_activate_time` returned None, implying
-                        // that the LastResult is NeverCompleted? the Some in
-                        // the second part of the tuple means this ran before,
-                        // so panic here.
-                        panic!("task is idle, but there's no activate time?!");
+                    // the task first reported that it completed, but now
+                    // reports that it has never completed
+                    (LastResult::Completed(_), LastResult::NeverCompleted) => {
+                        panic!("completed, then never completed?!");
                     }
 
-                    (Some(current_activation), Some(last_activation)) => {
-                        // the task is idle, it started ok, and it was
-                        // previously activated: compare times to make sure we
-                        // didn't observe the same BackgroundTask object
-                        if current_activation > last_activation {
+                    (LastResult::Completed(a), LastResult::Completed(b)) => {
```

Comment:
Nit: using variable names like `start`, `last` might read a little better (instead of `a`, `b`).

Reply:
Yeah, definitely: d1b02c4ef